javascript/javascript1/week4/ccformater.js

```diff
@@ -5,21 +5,25 @@
 let k = 0;
 
 function formatCreditCardNumber(number) {
-  const numbersIntoArray = String(number).split("");
-  let str = "";
-  for (i = 0; i < numbersIntoArray.length; i++) {
-    console.log(i);
-    if (i % 4 === 0 && i !== 0) {
-      //   numbersIntoArray.splice(numbersIntoArray[i + k], 0, " ");
-      str += " " + numbersIntoArray[i];
-      // k = k + 1;
-    } else {
-      str += numbersIntoArray[i];
+  if (typeof number !== "number") {
+    console.log("Please type numbers");
+  } else {
+    const numbersIntoArray = String(number).split("");
+    let str = "";
+    for (i = 0; i < numbersIntoArray.length; i++) {
```

Comment:
I see where you are going but you don't need to convert the string to an array to be able to loop over it.

This would work for example:
```
for (const char of str) {
  console.log(char);
}
```